@here/harp-map-controls/lib/CameraAnimationBuilder.ts

```diff
@@ -0,0 +1,178 @@
+    }
+
+    /**
+     * Sets the view to the defiend [[CameraPose]]
+     */
+    static setViewFromPose(mapView: MapView, pose: CameraPose) {
+        mapView.lookAt({
+            target: pose.geoCoordinate,
+            distance: pose.distance,
+            tilt: pose.direction.altitude,
+            heading: pose.direction.azimuth
+        });
+    }
+
+    /**
+     * Appends a [[ControlPoint]] to [[CameraKeyTrackAnimationOptions]]
+     *
+     * @param options
+     * @param point
+     */
+    static appendControlPoint(options: CameraKeyTrackAnimationOptions, point: ControlPoint) {
+        const timestamp = options.controlPoints[options.controlPoints.length - 1].timestamp + 10;
+        point.timestamp = timestamp;
+        options.controlPoints.push(point);
+    }
+
+    /**
+     * Adds a [[ControlPoint]] to the beginning of an [[CameraKeyTrackAnimationOptions]]
+     *
+     * @param options
+     * @param point
+     * @param prependTime The time the animation from the inserted key to the next should take
+     */
+    static prependControlPoint(
+        options: CameraKeyTrackAnimationOptions,
+        point: ControlPoint,
+        prependTime?: number
+    ) {
+        prependTime = prependTime !== undefined ? prependTime : 10;
+        for (const controlPoint of options.controlPoints) {
+            controlPoint.timestamp += prependTime;
+        }
+        point.timestamp = 0;
+        options.controlPoints.unshift(point);
+    }
+
+    /**
+     *
+     * Creates Options for a Bow Animation from the start to the target [[ControlPoint]]
+     *
+     * @param startControlPoint
+     * @param altitude
+     * @param targetControlPoint
+     * @param duration The duration of the Animation, defaults to 10 seconds
+     */
+    static createBowFlyToOptions(
+        mapView: MapView,
+        startControlPoint: ControlPoint,
+        targetControlPoint: ControlPoint,
+        altitude?: number,
+        duration = 10
+    ): CameraKeyTrackAnimationOptions {
+        const controlPoints: ControlPoint[] = [startControlPoint];
+        const maxAltitude =
+            2 *
+            mapView.camera.position.distanceTo(
+                mapView.projection.projectPoint(targetControlPoint.geoCoordinate)
+            );
+
+        const sphericalCoordinates = MapViewUtils.extractSphericalCoordinatesFromLocation(
+            mapView,
+            mapView.camera,
+            targetControlPoint.geoCoordinate
+        );
+        targetControlPoint.direction.azimuth =
+            180 - THREE.MathUtils.radToDeg(-sphericalCoordinates.azimuth);
+
+        const midCoord0 = this.geoCoordLerp(
+            startControlPoint.geoCoordinate,
+            targetControlPoint.geoCoordinate,
+            0.5
+        );
+        const midPoint0 = new ControlPoint({
+            geoCoordinate: midCoord0,
+            distance: maxAltitude,
+            timestamp: duration / 2,
+            altitude: 0,
+            azimuth: THREE.MathUtils.lerp(
```

Comment:
This is not proper mid angle. If start is 1 and end is 359, it will attempt to rotate almost full circle to get to mid point.

Please use `MathUtils.interpolateAnglesDeg` from geoutils

Reply:
done